dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/Event.scala

```diff
@@ -0,0 +1,73 @@
+package org.bitcoins.dlc.oracle
+
+import org.bitcoins.commons.jsonmodels.dlc.SigningVersion
+import org.bitcoins.crypto.{FieldElement, SchnorrDigitalSignature, SchnorrNonce}
+import org.bitcoins.dlc.oracle.storage._
+
+sealed trait Event {
+
+  /** The nonce the oracle is committing to for this event */
+  def nonce: SchnorrNonce
+
+  /** The name given to this event, may be a URI */
+  def eventName: String
+
+  /** The total number of outcomes */
+  def numOutcomes: Long = outcomes.size
+
+  /** The version of signing for this event */
+  def signingVersion: SigningVersion
+
+  /** A signature by the oracle of the hash of nonce */
+  def commitmentSignature: SchnorrDigitalSignature
```

Comment:
Should probably sign the eventName with the nonce

Reply:
Already was, will fix comment